baseplate/integration/thrift/__init__.py

```diff
@@ -42,82 +40,103 @@ class RequestContext(object):
+                raise
+            except TException:
+                # this is an expected exception, as defined in the IDL
+                span.finish()
+                raise
+            except:
+                # the handler crashed!
+                span.finish(exc_info=sys.exc_info())
+                raise
+            else:
+                # a normal result
+                span.finish()
+                return result
+        return call_with_context
+
+
+def _extract_trace_info(headers):
+    extracted_values = TraceInfo.extract_upstream_header_values(TRACE_HEADER_NAMES, headers)
+    flags = extracted_values.get("flags", None)
+    return TraceInfo.from_upstream(
+        int(extracted_values["trace_id"]),
+        int(extracted_values["parent_span_id"]),
+        int(extracted_values["span_id"]),
+        True if extracted_values["sampled"].decode("utf-8") == "1" else False,
+        int(flags) if flags is not None else None,
+    )
+
+
+def baseplateify_processor(processor, logger, baseplate, edge_context_factory=None):
+    """Wrap a Thrift Processor with Baseplate's span lifecycle.
+
+    :param thrift.Thrift.TProcessor processor: The service's processor to wrap.
+    :param logging.Logger logger: The logger to use for error and debug
+        logging.
     :param baseplate.core.Baseplate baseplate: The baseplate instance for your
         application.
     :param baseplate.core.EdgeRequestContextFactory edge_context_factory: A
         configured factory for handling edge request context.
 
     """
-    def __init__(self, logger, baseplate, edge_context_factory=None):
-        self.logger = logger
-        self.baseplate = baseplate
-        self.edge_context_factory = edge_context_factory
-
-    def getHandlerContext(self, fn_name, server_context):
-        context = RequestContext()
-
-        trace_info = None
-        headers = server_context.iprot.trans.get_headers()
-        try:
-            trace_info = self._get_trace_info(headers)
-            edge_payload = headers.get(b"Edge-Request", None)
-            if self.edge_context_factory:
-                edge_context = self.edge_context_factory.from_upstream(
-                    edge_payload)
-                edge_context.attach_context(context)
-            else:
-                # just attach the raw context so it gets passed on
-                # downstream even if we don't know how to handle it.
-                context.raw_request_context = edge_payload
-        except (KeyError, ValueError):
-            pass
-
-        trace = self.baseplate.make_server_span(
-            context,
-            name=fn_name,
-            trace_info=trace_info,
-        )
-
-        try:
-            peer_address, peer_port = server_context.getPeerName()
-        except AttributeError:
-            # the client transport is not a socket
-            pass
-        else:
-            trace.set_tag("peer.ipv4", peer_address)
-            trace.set_tag("peer.port", peer_port)
```

Comment:
looks like these tags aren't set anymore

Reply:
yeah, i should've called that out. apache thrift doesn't have the socket context passed down to the server like fbthrift does, i'll poke around but i don't think it'll be as easy to get these in :(